File: android/src/main/kotlin/com/entrig/plugin/NotificationEvent.kt
package com.entrig.plugin

data class NotificationEvent(
    val title: String?,
    val body: String?,
    val type: String?,
    val deeplink: String?,
    val data: Map<String, Any?>?
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "title" to title,
            "body" to body,
            "type" to type,
            "deeplink" to deeplink,
            "data" to data
        )
    }

}

File: lib/src/notification_event.dart
class NotificationEvent {
  NotificationEvent({
    required this.title,
    required this.body,
    this.type,
    this.deeplink,
    this.data,
  });

  String title;
  String body;
  String? type;
  String? deeplink;
  Map? data;

  factory NotificationEvent.fromMap(Map eventData) {
    return NotificationEvent(
      title: eventData['title'] ?? '',
      body: eventData['body'] ?? '',
      type: eventData['type'],
      deeplink: eventData['deeplink'],
      data: eventData['data'],
    );
  }
}
